e2etest/zt_resume_test.go

```diff
@@ -83,4 +85,54 @@ func TestResume_LargeGeneric(t *testing.T) {
 		},
 		EAccountType.Standard(),
 	)
-}
\ No newline at end of file
+}
+
+func TestResume_PublicSource_BlobTarget(t *testing.T) {
+	RunScenarios(
+		t,
+		// copy only instead of sync because single file sync is not entirely compatible with the test suite
+		// the destination file must exist for sync to happen (so, a copy must happen first)
+		// in addition,
+		eOperation.Copy()|eOperation.Resume(),
+		eTestFromTo.Other(common.EFromTo.BlobBlob(), common.EFromTo.BlobLocal()),
+		eValidate.Auto(),
+		params{
+			recursive:      true,
+			debugSkipFiles: []string{";"}, // skip the only file is ;
+		},
+		nil,
+		testFiles{
+			defaultSize:  "1K",
+			sourcePublic: azblob.PublicAccessBlob,
+			objectTarget: "a.txt",
+
+			shouldTransfer: []interface{}{
+				f("a.txt"),
+			},
+		},
+		EAccountType.Standard(),
+	)
+}
+
+func TestResume_PublicSource_ContainerTarget(t *testing.T) {
+	RunScenarios(
+		t,
+		eOperation.CopyAndSync()|eOperation.Resume(),
+		eTestFromTo.Other(common.EFromTo.BlobBlob(), common.EFromTo.BlobLocal()),
+		eValidate.Auto(),
+		params{
+			recursive:      true,
+			debugSkipFiles: []string{"a.txt"}, // skip the only file is ;
+		},
+		nil,
+		testFiles{
+			defaultSize:  "1K",
+			sourcePublic: azblob.PublicAccessContainer,
+
+			shouldTransfer: []interface{}{
+				f("a.txt"),
```

Comment:
Let's add a virtual directory too. Obviously, the container in this case will be public.

```golang
folder("testfolder1")
file("testfolder1/a.txt")
```